docs-src/go/generated/add-replay-test-to-background-check-workflow.md

```diff
@@ -0,0 +1,33 @@
+---
+id: add-replay-test-to-background-check-workflow
+title: Add a Replay test
+sidebar_label: Add Replay test
+description: Define the code needed to run a Worker Process in Go.
+tags:
+- testing
+- replay test
+- replayer
+---
+
+<!-- DO NOT EDIT THIS FILE DIRECTLY.
+THIS FILE IS GENERATED from https://github.com/temporalio/documentation-samples-go/blob/port_replay_test_dacx/backgroundcheck_replay/tests/backgroundcheck_test.go. -->
+
+Add the Replay test to the set of application tests.
+The Replayer is available from the `go.temporal.io/sdk/worker` package in the SDK.
+Register the Workflow Definition and then specify an existing Event History to compare to.
+
+Run the tests in the test directory (`go test`).
+If the Workflow Definition and the Event History are incompatible then the test fails.
+
+<div class="copycode-notice-container"><div class="copycode-notice"><img data-style="copycode-icon" src="/icons/copycode.png" alt="Copy code icon" /> Sample application code information <img id="i-11d063aa-5a7c-456f-acab-35f2b1c66f16" data-event="clickable-copycode-info" data-style="chevron-icon" src="/icons/chevron.png" alt="Chevron icon" /></div><div id="copycode-info-11d063aa-5a7c-456f-acab-35f2b1c66f16" class="copycode-info">The following code sample comes from a working and tested sample application. The code sample might be abridged within the guide to highlight key aspects. Visit the source repository to <a href="https://github.com/temporalio/documentation-samples-go/blob/port_replay_test_dacx/backgroundcheck_replay/tests/backgroundcheck_test.go">view the source code</a> in the context of the rest of the application code.</div></div>
+
+```go
+// TestReplayWorkflowHistoryFromFile tests for Event History compatability.
+func (s *UnitTestSuite) TestReplayWorkflowHistoryFromFile() {
+	// Create a new Replayer
+	replayer := worker.NewWorkflowReplayer()
+	// Register the Workflow with the Replayer
+	replayer.RegisterWorkflow(workflows.BackgroundCheck)
+	// Compare the current Workflow code against the existing Event History
+	err := replayer.ReplayWorkflowHistoryFromJSONFile(nil, "backgroundcheck_workflow_event_history.json")
```

Comment:
I think it's important to have this example show fetching the history from server and replaying it in one go rather than reading a file from disk, as the most typical and important setup for users is replaying histories in CI that are open right now or recently finished. 

Stored histories can be useful for verification of changes over long periods of time, etc, but generally the easiest pattern for users is going to be to query visibility for some recent workflows and then fetch and replay those all in one go

Reply:
👍 We will definitely showcase that at some point.
Starting off with a baby step in this iteration.